tests/code_update.py

```diff
@@ -352,38 +352,78 @@ def test_proposal_invalidation(network, args):
+                        if args.http2
+                        else infra.interfaces.AppProtocol.HTTP1,
+                    )
+                }
+            )
+        ],
+        args.binary_dir,
+        args.debug_nodes,
+        args.perf_nodes,
+        pdb=args.pdb,
+    ) as network:
+        network.start_and_open(args)
 
-    while time.time() < end_time and not os.path.exists(args.snp_secondary_acis_path):
         LOG.info(
-            f"({time.time() - start_time}) Waiting for SNP secondary IP addresses file at: ({args.snp_secondary_acis_path}) to be created"
+            f"Secondary ACI information expected at: {args.snp_secondary_acis_path}"
         )
-        time.sleep(10)
-
-    if os.path.exists(args.snp_secondary_acis_path):
-        LOG.info("SNP secondary IP addresses file created")
-        with open(args.snp_secondary_acis_path, "r", encoding="utf-8") as f:
-            secondary_acis = [
-                tuple(secondary_aci.split(" "))
-                for secondary_aci in f.read().splitlines()
-            ]
-            for secondary_name, secondary_ip in secondary_acis:
-                LOG.info(
-                    f'Secondary ACI with name "{secondary_name}" has IP: {secondary_ip}'
-                )
-
-    else:
-        LOG.error("SNP secondary IP addresses file not created before timeout")
+        if args.snp_secondary_acis_path is None:
+            LOG.warning(
+                "Skipping test snp secondary deployment as no target secondary ACIs specified"
+            )
+            return network
+
+        timeout = 60 * 60  # 60 minutes
+        start_time = time.time()
+        end_time = start_time + timeout
+
+        while time.time() < end_time and not os.path.exists(
+            args.snp_secondary_acis_path
+        ):
+            LOG.info(
+                f"({time.time() - start_time}) Waiting for SNP secondary IP addresses file at: ({args.snp_secondary_acis_path}) to be created"
+            )
+            time.sleep(10)
+
+        if os.path.exists(args.snp_secondary_acis_path):
+            LOG.info("SNP secondary IP addresses file created")
+            with open(args.snp_secondary_acis_path, "r", encoding="utf-8") as f:
+                secondary_acis = [
+                    tuple(secondary_aci.split(" "))
+                    for secondary_aci in f.read().splitlines()
+                ]
+                for secondary_name, secondary_ip in secondary_acis:
+                    LOG.info(
+                        f'Secondary ACI with name "{secondary_name}" has IP: {secondary_ip}'
+                    )
+                    new_node = network.create_node(f"aci://{secondary_ip}")
+                    network.join_node(new_node, args.package, args, timeout=3)
+                    network.trust_node(new_node, args)
+                    LOG.info(
+                        f"Secondary ACI with name {secondary_name} joined the network"
+                    )
+
```

Comment:
Can you check here that you can create a client to `new_node` and issue (any) requests to it?

Reply:
https://github.com/microsoft/CCF/pull/4943/commits/f35fe064efd1d832d23a64c62da497d3fdc808c4